### ISSUE TITLE
list-view doesn't support Ember 1.11.beta-5

### ISSUE THREAD

**Issue by tim-evans** [emberjs/list-view#209]
Morph seems to have changed its API between 1.10 and 1.11.

The error is that `this.lastNode` is null. (last line in the `insertBefore` call) I'm not sure how to fix this.

```javascript
Morph.prototype.insertBeforeMorph = function(morph, referenceMorph) {
      if (referenceMorph && referenceMorph.parentMorph !== this) {
        throw new Error('The morph before which the new morph is to be inserted is not a child of this morph.');
      }

      morph.parentMorph = this;

      var parentNode = this.firstNode.parentNode;

      insertBefore(
        parentNode,
        morph.firstNode,
        morph.lastNode,
        referenceMorph ? referenceMorph.firstNode : this.lastNode.nextSibling
      );
```

**Comment by danielspaniel** [comment]
@tim-evans, what is the last version of ember that supported list-view  for you?  
Curious, because I am trying to upgrade to ember 1.11 - anything, and am blocked by this one.

**Comment by tim-evans** [comment]
I was on 1.11-canary, which worked for me. I'm not sure what happened between the releases

**Comment by danielspaniel** [comment]
How do you set the ember setting in bower.json to use the canary release?

**Comment by tim-evans** [comment]
Hrrm, you use `bower install ember#canary`, but that would leave you at 1.12 canary rather than 1.11

**Comment by danielspaniel** [comment]
Got it .. will give that a try

**Comment by danielspaniel** [comment]
Using ```bower install ember#canary``` I am still getting that "nextSibling"  error  ( ember 1.12 canary ) .. crazy .. how did you get ember 1.11-canary .. ?  
 Seems hard to believe that this would work in 1.11 canary, then break again in 1.12 ?? 
 can you send you bower.json and package.json?

**Comment by taras** [comment]
@danielspaniel it's happened before, so I wouldn't be surprised :)

**Comment by stefanpenner** [comment]
@taras and me resolved many of the issues, several stream related ones remain

**Comment by danielspaniel** [comment]
thanks @stefanpenner for update .. should I wait for 0.1.0 release to start using this with ember 1.11

**Comment by marcemira** [comment]
:+1: Any news on 1.11 compat?

**Comment by stefanpenner** [comment]
@marcemira https://github.com/emberjs/list-view/pull/219 is the WIP PR, it will likely work for your needs, but we still have some failing tests to fix. Unfortunately we haven't had a chance to bring those to completion yet.

**Comment by mrinterweb** [comment]
I checked out the pr-219-fix branch and tried to build. The readme instructions of `npm run build-all` result in "missing script: build-all". `npm run build` produces something that I don't believe to be usable. Can anyone suggest a way to build list-view? I'm trying to use it with ember 1.11 and ember-cli 0.2.2.

**Comment by stefanpenner** [comment]
@mrinterweb you don't need to build that PR, it is an addon, it should just work as other addons

cc @taras

**Comment by mrinterweb** [comment]
@stefanpenner for ember-cli should I run `ember install:addon list-view`? If so, how would I specify the pr-219-fix branch? I think I'm missing something here in terms of how to get list view working with ember-cli.

**Comment by stefanpenner** [comment]
@mrinterweb npm can download git repo's and branches aswell

**Comment by taras** [comment]
@mrinterweb you should be able to install list-view from that branch with `npm install --save "emberjs/list-view#pr-219-fix"`

**Comment by taras** [comment]
Here is an [example app](embersherpa/list-view-example-app) that shows it working and [package.json](https://github.com/embersherpa/list-view-example-app/blob/master/package.json#L37)

**Comment by mrinterweb** [comment]
Thank you @taras. That example is useful. Things 

Kind of unrelated question: What should be imported when defining a view that extends Ember.ListView?
I tried:
```javascript
import ListView from "list-view/list-view";
MyView = ListView.extend({});
```
"Uncaught TypeError: Cannot read property 'extend' of undefined"

Thanks to your example, I learned about the ember-list component. Everything is working for me except extending ListView (which I don't really need to do).

**Comment by danielspaniel** [comment]
If you put this line in Brocfile.js 

```js
app.import('bower_components/ember-list-view/list-view.js');
```
right below here:
```js
// Use `app.import` to add additional libraries to the generated
// output files.
```
Ember.ListView will be available in your application to use as you described. You don't need to import it when your about to use it in other words.

**Comment by davidsteinberger** [comment]
@taras: Thanks for the PR and the example. What would be the best way to extend the list-view used by the `ember-list` component? I tried using an initializer, however that did not work out.

**Comment by tim-evans** [comment]
Closed by #225
